src/shadow-embed.js

```diff
@@ -48,13 +48,24 @@ const CSS_SELECTOR_END_REGEX = /[^\-\_0-9a-zA-Z]/;
  * @param {!Element} hostElement
  * @return {!ShadowRoot}
  */
-export function createShadowRoot(hostElement) {
+export function createShadowRoot(hostElement, isCriticalPath) {
   const existingRoot = hostElement.shadowRoot || hostElement.__AMP_SHADOW_ROOT;
   if (existingRoot) {
     existingRoot./*OK*/innerHTML = '';
     return existingRoot;
   }
 
+  if (isCriticalPath) {
```

Comment:
We maybe able to live with this to start with, but probably worth to create `AmpDocShell` subclass of `AmpDoc`? WDYT?